### PR TITLE
Feature/explicit read

### PR DESCRIPTION
**Pull request by martintaraz**
Modifications on the python source code for working with an explicit given read schemata for decoding avro messages
for tesing purposes, the integration where ran against a docker kafka build

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,34 +5,17 @@ matrix:
     language: python
     dist: trusty
     python: "2.7"
-    env: LD_LIBRARY_PATH="$PWD/tmp-build/lib" LIBRDKAFKA_VERSION=v0.11.5
-  # Source package verification with Python 3.6 and librdkafka v0.11.5
+    services: docker
```
Comment (rnpridgeon) [inline .travis.yml]: All changes to .travis.yml should be removed from this commit

File: examples/integration_test.py [hunk 2]
```diff
@@ -21,7 +21,6 @@
 """ Test script for confluent_kafka module """
 
 import confluent_kafka
-from confluent_kafka import admin
```
Comment (rnpridgeon) [inline examples/integration_test.py]: Please remove any integration test changes not related to the pull request itself. This helps to keep the commit history clean. If you wish to submit a PR which cleans up formatting, white space and imports I'd be happy to review and merge however :)

**Comment by fkaufer** [comment]
> I have to ask though what is the use case for setting the reader schema here. On the surface this seems like a way to handle forward-compatible changes. If that is the case allowing the Schema Registry enforce compatibility constraints seems like a more robust solution for this situation.

@rnpridgeon No, it's about actually taking advantage of Avro's back-/forward compatibility features for which you need to be able to separate writer and reader schema. Before the change, the reader schema was (implicitly) hard-coded to the writer schema (of the respective message), now the consumer can pin a reader schema with the fastavro deserializer doing back or forward schema migration to the reader schema. The Schema Registry only enforces that the sequence of published Avro schemas for a certain topic complies with back-/forward compatibility constraints.

**Comment by fkaufer** [comment]
> @fkaufer you are absolutely correct, I had only considered the ability to parse the message. Sorry about that. Other than the .travis.yml changes it LGTM. If you could back those out it would be greatly appreciated.
Reverted the .travis.yml and other non-related changes in https://github.com/confluentinc/confluent-kafka-python/pull/469

**Comment by rnpridgeon** [comment]
Fixed with #470

**Pull request closed without merging**